libdnf/repo/config_repo.cpp

```diff
@@ -87,7 +87,7 @@ class ConfigRepo::Impl {
     OptionChild<OptionString> user_agent{main_config.user_agent()};
     OptionChild<OptionBool> countme{main_config.countme()};
     OptionEnum<std::string> failovermethod{"priority", {"priority", "roundrobin"}};
-    OptionBool build_cache{false};
+    OptionBool build_cache{main_config.build_cache()};
```

Comment:
I think we want the repository option to be linked to the global option - `OptionChild`. Not a copy.
`OptionChild<OptionBool> build_cache{main_config.build_cache()};`

Reply:
I also recommend to use `false` for a commandline repository. What about to modify commandline repository at the time of creation?
```
RepoWeakPtr RepoSack::get_cmdline_repo() {
    if (!cmdline_repo) {
        cmdline_repo = std::make_unique<Repo>(base, CMDLINE_REPO_NAME, Repo::Type::COMMANDLINE);
    }

    return cmdline_repo->get_weak_ptr();
}
```

Reply:
I guess we should switch it off also for the @System repo.

Reply:
In future when we will check correctly validity of system repository cache we can enable the caching.

Reply:
I've added a TODO for it.